File: abc_dz_2/abc_dz2_source/classes/truck.cpp
//
// Created by Дмитрий Татьянченко on 18.10.2021.
//

#include "truck.h"

Truck::Truck(int fuel_tank_capacity, double fuel_consumption, int carrying_capacity) :
        Vehicle(fuel_tank_capacity, fuel_consumption) {

    this->type_ = (Vehicle::TRUCK);

    if (carrying_capacity < 0) {
        throw std::invalid_argument("carrying capacity < 0");
    }

    this->carrying_capacity_ = carrying_capacity;
}

std::string Truck::toString() {
    return "Truck: " + this->Vehicle::toString() +
           " carrying capacity: " + std::to_string(this->carrying_capacity_);
}

File: abc_dz_3/abc_dz3_source/vehicle/truck.py
from vehicle.vehicle import *


class Truck(Vehicle):
    def __init__(self, fuel_tank_capacity, fuel_consumption, carrying_capacity):
        super().__init__(fuel_tank_capacity, fuel_consumption)
        self.__carrying_capacity = carrying_capacity

    def to_string(self):
        return "TRUCK " + super().to_string() + " carrying capacity (tons): " + str(self.__carrying_capacity)

